ansi/buffer_test.go

```diff
@@ -33,10 +34,51 @@ func Benchmark_PrintableRuneWidth(b *testing.B) {
 func Test_Truncate(t *testing.T) {
 	t.Parallel()
 
-	s := "\x1B[38;2;249;38;114m你\x1B[7m好\x1B[0m"
+	tests := []struct {
+		in            string
+		out           string
+		width         int
+		expectedWidth int
+	}{
+		{
+			"\x1B[38;2;249;38;114m你\x1B[7m好\x1B[0m",
+			"\x1B[38;2;249;38;114m你\x1B[7m\x1B[0m",
+			2,
+			2,
+		},
+		{
+			"\x1B[38;2;249;38;114m你\x1B[7m好\x1B[0m",
+			"\x1B[38;2;249;38;114m\x1B[0m",
+			1,
+			0,
+		},
+		{
+			"It’s me!",
+			"It’s me!",
+			10,
+			8,
+		},
+		{
+			"It’s \x1B[7mme!",
+			"It’s \x1B[7m\x1B[0m",
+			5,
+			5,
+		},
+	}
 
-	if n := PrintableRuneWidth(Truncate(s, 2, "")); n != 2 {
-		t.Fatalf("width should be 2, got %d", n)
+	i := 0
+	for _, tt := range tests {
```

Comment:
Would `for i, tt := range tests {` be better?

Reply:
Good call. I'll update.